lib/puppet/defaults.rb

```diff
@@ -939,26 +945,37 @@ def self.default_checksum_types
 EOT
     },
     :digest_algorithm => {
-        :default  => lambda { default_digest_alg },
+        :default  => lambda { default_digest_algorithm },
         :type     => :enum,
-        :values => ["md5", "sha256", "sha384", "sha512", "sha224"],
-        :desc     => 'Which digest algorithm to use for file resources and the filebucket.
-                      Valid values are md5, sha256, sha384, sha512, sha224. Default is md5.',
+        :values   => valid_digest_algorithms,
+        :desc     => "Which digest algorithm to use for file resources and the filebucket.
+                      Valid values are #{valid_digest_algorithms.join(', ')}. Default is
+                      #{valid_digest_algorithms.first}.",
     },
     :supported_checksum_types => {
-      :default => lambda { default_checksum_types },
+      :default => lambda { default_file_checksum_types },
       :type    => :array,
-      :desc    => 'Checksum types supported by this agent for use in file resources of a
-                   static catalog. Values must be comma-separated. Valid types are md5,
-                   md5lite, sha256, sha256lite, sha384, sha512, 
-		   sha1, sha1lite, sha224, mtime, ctime.',
+      :desc    => "Checksum types supported by this agent for use in file resources of a
+                   static catalog. Values must be comma-separated. Valid types are
+                   #{Puppet.valid_file_checksum_types.join(', ')}. Default is
+                   #{Puppet.default_file_checksum_types.join(', ')}.",
       :hook    => proc do |value|
         values = munge(value)
-        valid   = ['md5', 'md5lite', 'sha256', 'sha256lite', 'sha384', 'sha512', 'sha224', 'sha1', 'sha1lite', 'mtime', 'ctime']
-        invalid = values.reject {|alg| valid.include?(alg)}
+
+        if Puppet::Util::Platform.fips_enabled?
+          invalid = values.find { |v| v == 'md5' || v == 'md5lite' }
```

Comment:
Isn't this effectively duplicating the invalid check below, since `Puppet.valid_checksum_types` won't contain `md5`, or `md5lite` when `Puppet::Util::Platform.fips_enabled?` is true? (Eg: `invalid = values - Puppet.valid_checksum_types`)